src/plugins/data_source_management/public/components/data_source_table/data_source_table.test.tsx

```diff
@@ -429,4 +433,37 @@ describe('DataSourceTable', () => {
       mockedContext.application.capabilities = capabilities;
     });
   });
+
+  describe('should handle datasources with empty description correctly', () => {
+    beforeEach(async () => {
+      spyOn(utils, 'getDataSources').and.returnValue(
+        Promise.resolve(getMappedDataSourcesWithEmptyDescription)
+      );
+      spyOn(uiSettings, 'get$').and.returnValue(new BehaviorSubject('test1'));
+      await act(async () => {
+        component = await mount(
+          wrapWithIntl(
+            <DataSourceTable
+              history={history}
+              location={({} as unknown) as RouteComponentProps['location']}
+              match={({} as unknown) as RouteComponentProps['match']}
+            />
+          ),
+          {
+            wrappingComponent: OpenSearchDashboardsContextProvider,
+            wrappingComponentProps: {
+              services: mockedContext,
+            },
+          }
+        );
+      });
+      component.update();
+    });
+
+    it('should render normally', () => {
+      expect(() => component).not.toThrow();
+      expect(component).toMatchSnapshot();
```

Comment:
Can we assert on the elements instead of snapshot? Like we can find 3 rows or data sources even if some's description is null or undefined.

Reply:
Done, added assertion for the rows and placeholders